crates/script/src/transaction.rs

```diff
@@ -159,8 +160,13 @@ impl ScriptTransactionBuilder {
 
         if !self.transaction.is_fixed_gas_limit {
             if let Some(unsigned) = self.transaction.transaction.as_unsigned_mut() {
+                let mut gas = result.gas_used;
+                // if this is a 7702 tx, add the per-account auth costs
+                if let Some(auth_list) = unsigned.authorization_list.as_ref() {
+                    gas += auth_list.len() as u64 * PER_EMPTY_ACCOUNT_COST;
+                }
                 // We inflate the gas used by the user specified percentage
-                unsigned.gas = Some(result.gas_used * gas_estimate_multiplier / 100);
+                unsigned.gas = Some(gas * gas_estimate_multiplier / 100);
```

Comment:
looks like this is required because we never set the auth list when simulating tx here https://github.com/foundry-rs/foundry/blob/c4a31a624874ab36284fca4e48d2197e43a62fbe/crates/script/src/simulate.rs#L121

if we set it correctly the gas used would contain the usage for the auth list as well removing the need to handle it manually

Reply:
Thanks - you were right about letting revm handle the 7702 gas. I've removed the manual gas calc and properly set the authorization list in the tx env before simulation.